.github/workflows/register_repositories.py

```diff
@@ -28,23 +53,33 @@ def parse_line(line, original_config, repos={'remove': [], 'add': []}):
             x = item.strip().split(':')
             info[x[0]] = x[1]
 
-        if split_on == '/add-repo':
-            if info['repo'] not in set(config[info['campaign']]['repos']):
-                config[info['campaign']]['repos'].append(info['repo'])
-                repos['add'].append(info['repo'])
+        valid, error_messages = validate_repo_campaign_info(info, config)
 
-        else:
-            try:
-                config[info['campaign']]['repos'].remove(info['repo'])
-            except ValueError:
-                pass
-            finally:
-                repos['remove'].append(info['repo'])
+        if valid:
+            if split_on == add_cmd:
+                if info['repo'] not in set(config[info['campaign']]['repos']):
+                    config[info['campaign']]['repos'].append(info['repo'])
+                    repos['add'].append(info['repo'])
+
+            else:
+                try:
+                    config[info['campaign']]['repos'].remove(info['repo'])
+                except ValueError:
+                    error_messages.append(
+                        f"\n  - Unable to remove the following repo: {info['repo']} because it doesn't exist in the list of repos: {config[info['campaign']]['repos']} in the {info['campaign']} campaign."
```

Comment:
```suggestion
                        f"\n  - Unable to remove the repo {repr(info['repo'])} because it doesn't exist in the list of repos ({', '.join(repr(i) for i in config[info['campaign']]['repos'])}) of the {repr(info['campaign'])} campaign."
```